src/components/DataPanel/DataPanel.tsx

```diff
@@ -53,15 +51,17 @@ function DataPanel(props: Props) {
   }, [props.controlOption, checkList]);
 
   useEffect(() => {
-    setIsHidden(updateVisible);
     window.localStorage.setItem("checkList", JSON.stringify(checkList));
+    const isHidden = updateVisible();
     const newCheckList = [...checkList].map(
       (checkBox: CheckBox, idx: number) => {
         checkBox.isHidden = isHidden[idx];
         return checkBox;
       }
     );
-    props.updateCheckList(newCheckList);
+    setTimeout(() => {
+      props.updateCheckList(newCheckList);
+    }, 0);
```

Comment:
After removing `setTimeout`, lucky pick still works.
I didn't figure out why `setTimeout` with delay 0 is used.
May Steven explain what the purpose of `setTimeout`?

Reply:
I understand what you mean. Thank you!